src/txn/txn.c

```diff
@@ -978,8 +977,27 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
 
 				__wt_txn_op_set_timestamp(session, op);
 			} else {
-				WT_ERR(__wt_txn_resolve_prepared_op(
-				    session, op, true));
+				/*
+				 * Decrement the resolved update count as we
+				 * as we would either resolve at least 1 update
+				 * or have previously resolved the update linked
+				 * to this modification when we resolved the
+				 * prepared updates of a previous modification
+				 * in this txn.
+				 */
+				resolved_update_count--;
```

Comment:
I'm finding this hard to wrap my head around.  I *think* the idea is that we're adding to `resolved_update_count` inside `__wt_txn_resolve_prepared_op` for every update that gets resolved.  And here we're decrementing it for every op we visit.  So it should end up at zero?

But then we use the resulting count to increment a statistic, so if it's supposed to be zero at the end, I'm missing something.

Would it lose something to maintain two counts, `update_count` (which gets incremented here) and `resolved_update_count` that is incremented in `__wt_txn_resolve_prepared_op`, assert that `resolved_update_count >= update_count` in the loop, then at the end assert that `resolved_update_count == update_count`?

That way there is something meaningful to use for statistics, and I think the check that all updates get resolved at the end is important.